tee-worker/litentry/pallets/identity-management/src/lib.rs

```diff
@@ -293,7 +294,22 @@ pub mod pallet {
 		// get the whole IDGraph, sorted by `link_block` (earliest -> latest)
 		pub fn get_id_graph(who: &Identity) -> IDGraph<T> {
 			let mut id_graph = IDGraphs::iter_prefix(who).collect::<IDGraph<T>>();
-			id_graph.sort_by(|a, b| Ord::cmp(&a.1.link_block, &b.1.link_block));
+
+			// Initial sort to ensure a deterministic order with case where prime identity `link_block` might be equal another identity-to be linked
+			id_graph.sort_by(|a, b| {
+				let a_is_prime_identity = &a.0 == who;
+				let b_is_prime_identity = &b.0 == who;
+
+				match (a_is_prime_identity, b_is_prime_identity) {
```

Comment:
Thank you!

Maybe we want a more general one - _theoretically_ two non-prime identities could be linked in the same block still, imagine you have two back-to-back identity linking requests, there's a chance that they finally get fit into one sidechain block.

So just checking if it's prime identity is not sufficient.
We need to have some rules that can be applied to any identity.

It can be simple as long as it's deterministic, e.g. using string alphabetic order of identity.to_did() if their `link_block`s are equal